src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -56,52 +56,56 @@
 
 public class ExtensionsRunner {
+    private ExtensionSettings extensionSettings = readExtensionSettings();
+    private DiscoveryNode opensearchNode;
 
-    public static final String REQUEST_EXTENSION_ACTION_NAME = "internal:discovery/extensions";
-
-    private static ExtensionSettings extensionSettings = null;
-
-    static {
-        try {
-            extensionSettings = getExtensionSettings();
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-    }
+    public ExtensionsRunner() throws IOException {}
 
-    private static final Settings settings = Settings.builder()
+    private final Settings settings = Settings.builder()
         .put("node.name", extensionSettings.getExtensionname())
         .put(TransportSettings.BIND_HOST.getKey(), extensionSettings.getHostaddress())
         .put(TransportSettings.PORT.getKey(), extensionSettings.getHostport())
         .build();
-    private static final Logger logger = LogManager.getLogger(ExtensionsRunner.class);
-    public static final TransportInterceptor NOOP_TRANSPORT_INTERCEPTOR = new TransportInterceptor() {
+    private final Logger logger = LogManager.getLogger(ExtensionsRunner.class);
+    private final TransportInterceptor NOOP_TRANSPORT_INTERCEPTOR = new TransportInterceptor() {
     };
 
-    public ExtensionsRunner() throws IOException {}
-
-    public static ExtensionSettings getExtensionSettings() throws IOException {
+    private ExtensionSettings readExtensionSettings() throws IOException {
         File file = new File(ExtensionSettings.EXTENSION_DESCRIPTOR);
         ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
         ExtensionSettings extensionSettings = objectMapper.readValue(file, ExtensionSettings.class);
         return extensionSettings;
     }
 
+    private void setOpensearchNode(DiscoveryNode opensearchNode) {
+        this.opensearchNode = opensearchNode;
+    }
+
+    private DiscoveryNode getOpensearchNode() {
+        return opensearchNode;
+    }
+
     PluginResponse handlePluginsRequest(PluginRequest pluginRequest) {
-        logger.info("Handling Plugins Request");
+        logger.info("Registering Plugin Request received from OpenSearch");
         PluginResponse pluginResponse = new PluginResponse("RealExtension");
+        opensearchNode = pluginRequest.getSourceNode();
+        setOpensearchNode(opensearchNode);
         return pluginResponse;
     }
 
-    IndicesModuleResponse handleIndicesModuleRequest(IndicesModuleRequest indicesModuleRequest) {
-        logger.info("Indices Module Request");
+    IndicesModuleResponse handleIndicesModuleRequest(IndicesModuleRequest indicesModuleRequest, TransportService transportService) {
+        logger.info("Registering Indices Module Request received from OpenSearch");
         IndicesModuleResponse indicesModuleResponse = new IndicesModuleResponse(true, true, true);
+
+        // CreateComponent
+        DiscoveryNode opensearchNode = getOpensearchNode();
```

Comment:
I still think you need a log warning here if this has not been set. Last comment... will mark this as a suggestion but approve.